src/compiler/checker.ts

```diff
@@ -8835,11 +8855,13 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 }
 
                 if (isEntityName(node) || isEntityNameExpression(node)) {
+                    if (isDeclarationName(node)) {
+                        return node;
+                    }
                     const { introducesError, node: result } = trackExistingEntityName(node, context);
                     hadError = hadError || introducesError;
-                    if (result !== node) {
-                        return result;
-                    }
+                    // We should not go to child nodes of the entity name, they will not be acesible
+                    return result;

```

Comment:
Previously this would continue to walk the expression and end up testing things it shouldn't. For example, in `E.A` it would first see the entity name `E.A` decide that is ok because `E` is accessible and then proceed  to walk to `A` and `E` independently testing them again.